docs/blog/2021-01-19-frontend-asset-passthrough.md

```diff
@@ -0,0 +1,109 @@
+However, this means that during deploys, particularly when deploying to canary (which at Lyft is a particular subset of the production environment),
+there will be an intermediary state in which the frontend assets requested by the user may not exist on the hosts serving that request while the rollout is still progressing, since during deploys, there will briefly exist two different versions of the application taking traffic.
+
+Typically, organizations solve this problem by simply adding a CDN to their service architecture and serve static assets from there.
+However, we decided to go a different route, as the Clutch architecture and design philosophy values simplicity and avoiding adding new dependencies unless absolutely necessary. This simplicity is core to Clutch's values as we want to keep Clutch easy to deploy and configure and cloud provider agnostic, with as few external dependencies as possible.
+
+## Solution
+
+This leads us to our solution, which we call Frontend Asset Passthrough.
+How this works is simple: when a request for a static asset does not reside on disk, Clutch will fallback to a configured provider to look for the asset.
+
+Lets first take a look at the [proto configration](https://github.com/lyft/clutch/blob/890245e7d2a1bf91623a9e74b39f1083dbd5ea2c/api/config/gateway/v1/gateway.proto#L105-L119) and then will go into more detail.
+
+```protobuf
+message Assets {
+  message S3Provider {
+    string region = 1;
+    string bucket = 2;
+    string key = 3;
+  }
+
+  oneof provider {
+    S3Provider s3 = 1;
+  }
+}
+```
+
+The `assets` configuration resides in the `gateway` configuration.
+
+At Lyft we utilized AWS S3 as the provider of choice and have implemented this setup as an example,
+however, the Clutch protobuf model is extensible and can easily be extended to add additional providers for any given deployment.
+
+For S3 the configuration is simple, specify a `region`, `bucket`, and the `key` where the assets live.
+This does require you to also configure the `clutch.service.aws` service,
+which enables Clutch fetch these assets with S3 API's.
+
+```yaml
+gateway:
+  assets:
+    s3:
+      region: us-east-1
+      bucket: static
+      key: clutch
+
+services:
+  - name: clutch.service.aws
+    typed_config:
+      "@type": types.google.com/clutch.config.service.aws.v1.Config
+      regions:
+        - us-east-1
+```
+
+With this configured, if a user requests an asset that does not reside on disk the configured passthrough provider will be utilized.
+If the provider has the correct asset it will be served to the user.
+
+Let's look at the diagram below to demonstrate this.
+Our user makes a request, it goes through our load balancer, which in Lyft's case is Envoy.
+Envoy is load balancing all versions of Clutch that are currently deployed; in our example we have two Clutch versions deployed **v1** and **v2**.
```

Comment:
```suggestion
Envoy is load balancing all versions of Clutch that are currently deployed. In our example we have two Clutch versions deployed: **v1** and **v2**.
```

Reply:
also use backticks for version numbers instead of bold